parser.cpp

```diff
@@ -0,0 +1,84 @@
+#include <algorithm>
+#include <utility>
+#include "parser.h"
+
+const std::map<std::string, std::string> Parser::loadInput(std::istream &jsonFile){
+	std::string line;
+	std::string data = "";
+	while(getline(jsonFile, line)){
+		data += line;
+	}
+	return loadInput(data);
+}
+
+const std::map<std::string, std::string> Parser::loadInput(std::string data){
+    std::map<std::string, std::string> attributes;
+
+	bool isItAJsonFile = false;
+	if(data.length()>=5){
+		isItAJsonFile = data.substr(data.length()-5, 5) == ".json";
```

Comment:
Jaj, ilyet inkább ne! Akkor készíts egy különálló fv.-t, ami kifejezetten fájlt nyit meg, ez meg foglalkozzon egy string parzolásával.

Reply:
Javítva, a Character::parseUnit függvénye megpróbálja először megnyitni a fájlt, ha nem sikerül, stringként kezeli tovább a beírt adatot.